src/main/java/com/hartwig/platinum/pdl/PDLConversion.java

```diff
@@ -10,7 +11,7 @@
 
 public interface PDLConversion {
 
-    List<PipelineInput> apply(final PlatinumConfiguration configuration);
+    Map<String, Future<PipelineInput>> apply(final PlatinumConfiguration configuration);
```

Comment:
I think it would better to return a List<Supplier<PipelineInput>> here than making the futures part of the interface. The way it is now we add similar complexity to each implementation rather than handling it in a single place at submission time. Also the map isn't really required as the sample id can be resolved from the PipelineInput at submission time as well .

Reply:
See https://github.com/hartwigmedical/platinum/pull/46